### PR TITLE
Add ApplicationInsights to cosmosdb factory.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Remove old UA.

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spring/data/documentdb/DocumentDbFactory.java [hunk 1]
```diff
@@ -9,40 +9,53 @@
...
         Assert.hasText(host, "host must not be empty!");
         Assert.hasText(key, "key must not be empty!");
 
         final ConnectionPolicy policy = ConnectionPolicy.GetDefault();
 
-        String userAgent = ";" + USER_AGENT_SUFFIX;
-        if (isBiEnabled && GetHashMac.getHashMac() != null) {
-            userAgent += ";" + GetHashMac.getHashMac();
-        }
-        policy.setUserAgentSuffix(userAgent);
```
Comment (yungezz) [inline src/main/java/com/microsoft/azure/spring/data/documentdb/DocumentDbFactory.java]: please keep old ua. old ua is setting at service side, it provides telemetry which is not able to be collected by spring boot starters, like # of subs, maybe subs distribution per location, resource size etc. telemetry collected in ai is just a supplement.

Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/azure/spring/data/documentdb/DocumentDbFactory.java]: OK. I remove that as it seems duplicated. Will recover that.

File: src/main/java/com/microsoft/azure/spring/data/documentdb/DocumentDbFactory.java [hunk 2]
```diff
@@ -11,38 +11,72 @@
...
+            client.getConnectionPolicy().setUserAgentSuffix(this.getUserAgentSuffix(isBiEnabled));
+        }
+
         this.documentClient = client;
+        this.telemetryProxy = new TelemetryProxy(this.isTelemetryAllowed());
+        this.trackCustomEvent();
+    }
+
+    private boolean isTelemetryAllowed() {
+        // panli: will allow user to disable this from application.properties in another PR.
```
Comment (yungezz) [inline src/main/java/com/microsoft/azure/spring/data/documentdb/DocumentDbFactory.java]: configurable telemetry collecting is must for GDPR. i'll fwd you email from bi team about it. pls implement this switch.

File: src/main/resources/ApplicationInsights.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ApplicationInsights xmlns="http://schemas.microsoft.com/ApplicationInsights/2013/Settings" schemaVersion="2014-05-30">
+    <!-- The key from the portal: -->
+    <InstrumentationKey>0c3d848e-4923-45b8-bd72-95fd5ad5448d</InstrumentationKey>
```
Comment (yungezz) [inline src/main/resources/ApplicationInsights.xml]: different key for test and prod package

File: src/test/resources/ApplicationInsights.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ApplicationInsights xmlns="http://schemas.microsoft.com/ApplicationInsights/2013/Settings" schemaVersion="2014-05-30">
+    <!-- The key from the portal: -->
+    <InstrumentationKey>0c3d848e-4923-45b8-bd72-95fd5ad5448d</InstrumentationKey>
```
Comment (yungezz) [inline src/test/resources/ApplicationInsights.xml]: test key

File: src/main/resources/ApplicationInsights.xml (hunk 3 above)
Comment (Incarnation-p-lee) [reply src/main/resources/ApplicationInsights.xml]: Will fill that field from Bi Team.

File: src/test/resources/ApplicationInsights.xml (hunk 4 above)
Comment (Incarnation-p-lee) [reply src/test/resources/ApplicationInsights.xml]: This is test Key for now.

**Pull request merged**